src/Driver/KeepAliveProcessDriver.php

```diff
@@ -2,51 +2,58 @@
 
 namespace Resgen\Common\Proc\Driver;
 
-use Resgen\Common\Proc\ProcessControl;
 use Resgen\Common\Proc\EscapeProcessException;
+use Resgen\Common\Proc\ProcessControl;
+use Resgen\Common\Proc\SignalException;
+
+class KeepAliveProcessDriver implements ProcessControl {
+	use Logs;
+
+	/** @var bool If true, a SignalException will be thrown when the signal is handled */
+	protected $throwSignalExceptions = false;
 
-class KeepAliveProcessDriver implements ProcessControl
-{
-    use Logs;
+	/** @var int */
+	protected $lastMsgTime;
 
-    /** @var int */
-    protected $lastMsgTime;
+	/** @var int */
+	protected $startTime;
 
-    /** @var int */
-    protected $startTime;
+	/** @var int */
+	protected $timeSpentProcessing = 0;
 
-    /** @var int */
-    protected $timeSpentProcessing = 0;
+	/** @var bool */
+	protected $interrupt = false;
 
-    /** @var bool */
-    protected $interrupt = false;
+	public function __construct(SignalHandler $signalHandler) {
+		$this->startTime = microtime(true);
+		$signalHandler->bind($this, 'sigHandle');
```

Comment:
Initialize `$this->throwSignalExceptions = env('LUMEN_PROC_THOWS_SIGNAL', 'disabled') == 'enabled';`